src/diffusers/pipelines/stable_diffusion_xl/pipeline_stable_diffusion_xl.py

```diff
@@ -718,6 +718,67 @@ def disable_freeu(self):
         """Disables the FreeU mechanism if enabled."""
         self.unet.disable_freeu()
 
+    def enable_fused_qkv_projections(self, unet: bool = True, vae: bool = True):
+        """
+        Enables fused QKV projections. For self-attention modules, all projection matrices (i.e., query,
+        key, value) are fused. For cross-attention modules, key and value projection matrices are fused.
+
+        <Tip warning={true}>
+
+        This API is 🧪 experimental.
+
+        <Tip>
+
+        Args:
+            unet (`bool`, defaults to `True`): To apply fusion on the UNet.
+            vae (`bool`, defaults to `True`): To apply fusion on the VAE.
+        """
+        from ...models.attention_processor import FusedAttnProcessor2_0
+
+        self.fusing_unet = False
+        self.fusing_vae = False
+
+        if unet:
+            self.fusing_unet = True
+            self.unet.enable_fused_qkv_projections(device=self.device, dtype=self.dtype)
+            self.unet.set_attn_processor(FusedAttnProcessor2_0())
```

Comment:
I don't think it makes sense to add variants of this attention processor (xformers, PT 1.x, etc.) for now as that way we can push for pure native PT 2.0. Also, it tends to show benefit when you use `torch.compile()`. So, either way, keeping it to PT 2.0 only, for now, makes more sense to me.